docs/zh/guide/advanced/expand.md

```diff
@@ -197,3 +199,277 @@ public enum FrameworkEnum {
+package com.github.testi;
+
+import com.ly.doc.builder.ProjectDocConfigBuilder;
+import com.ly.doc.model.ApiDoc;
+import com.ly.doc.model.ApiSchema;
+import com.ly.doc.model.annotation.FrameworkAnnotations;
+import com.ly.doc.template.IDocBuildTemplate;
+import com.thoughtworks.qdox.model.JavaClass;
+
+import java.util.Collection;
+
+/**
+ * QuarkusDocBuildTemplate.
+ *
+ * @author test
+ * @version 1.0.0
+ * @since 2024-07-02 13:43:50
+ */
+public class QuarkusDocBuildTemplate implements IDocBuildTemplate<ApiDoc> {
+
+    /**
+     * render api
+     *
+     * @param projectBuilder   ProjectDocConfigBuilder
+     * @param candidateClasses candidate classes
+     * @return api ApiSchema
+     */
+    @Override
+    public ApiSchema<ApiDoc> renderApi(ProjectDocConfigBuilder projectBuilder, Collection<JavaClass> candidateClasses) {
+        return null;
+    }
+
+    /**
+     * support framework.
+     *
+     * @param framework framework
+     * @return boolean
+     */
+    @Override
+    public boolean supportsFramework(String framework) {
+        // 匹配
+        return "Quarkus".equalsIgnoreCase(framework);
+    }
+
+    /**
+     * registered annotations.
+     *
+     * @return registered annotations
+     */
+    @Override
+    public FrameworkAnnotations registeredAnnotations() {
+        return null;
+    }
+
+    /**
+     * is entry point.
+     *
+     * @param javaClass            javaClass
+     * @param frameworkAnnotations frameworkAnnotations
+     * @return is entry point
+     */
+    @Override
+    public boolean isEntryPoint(JavaClass javaClass, FrameworkAnnotations frameworkAnnotations) {
+        return false;
+    }
+}
+
+```
+
+然后在 `resources/META-INF/services/com.ly.doc.spi.DocBuildTemplate` 文件中添加实现类的全类名。如果是`WebSocket`文档则在`resources/META-INF/services/com.ly.doc.template.IWebSocketDocBuildTemplate` 文件中添加实现类的全类名。
+
+#### 安装或部署
+将项目打包成`jar`包，并安装到本地仓库或发布到远程仓库。
+
+#### 使用新添加的框架解析
+调整 `smart-doc-maven-plugin` 插件依赖配置，新增上述项目的依赖：
+```xml
+      <plugin>
+        <groupId>com.ly.smart-doc</groupId>
+        <artifactId>smart-doc-maven-plugin</artifactId>
+        <version>[最新版]</version>
+        <configuration>
+          <!--指定生成文档的使用的配置文件-->
+          <configFile>./src/main/resources/smart-doc.json</configFile>
+          <!--指定项目名称-->
+          <projectName>测试</projectName>
+        </configuration>
+        <dependencies>
+        <dependency>
+            <!--引入上一步安装或部署的新增模块-->
+            <groupId>com.github.test</groupId>
+            <artifactId>smart-doc-extend</artifactId>
+            <version>1.0-SNAPSHOT</version>
+        </dependency>
+    </dependencies>
+      </plugin>
+```
+
+在项目中使用 `smart-doc` 时配置所使用的框架名称。默认情况下，`smart-doc` 使用 `Spring`，因此新加的框架需在配置文件中指定：
+`smart-doc.json` 文件中修改 `framework` 配置：
+```json
+{
+  "serverUrl": "http://127.0.0.1",
+  "isStrict": false,
+  "allInOne": true,
+  "outPath": "D://md2",
+  "framework": "quarkus"
+}
+```
+开发流程就是这样，主要的难点在于`IDocBuildTemplate`的实现处理。
+
+### 方案二：集成式开发
+集成依赖进行扩展
```

Comment:
This approach is very poor in terms of universality and maintainability from the perspective of research and development infrastructure construction. It can be deleted. The philosophy of the smart-doc community is also to guide users to follow best practices.